test/functional/product/list_test.rb

```diff
@@ -0,0 +1,78 @@
+require File.join(File.dirname(__FILE__), '../test_helper')
+require File.join(File.dirname(__FILE__), 'product_helpers')
+
+describe "listing products" do
+  include ProductHelpers
+
+  before do
+    @cmd = %w(product list)
+  end
+
+  let(:org_id) { 1 }
+  let(:subscription_id) { 1 }
+  let(:sync_plan_id) { 1 }
+  let(:empty_response) do
+    {
+      "total" => 0,
+      "subtotal" => 0,
+      "page" => "1",
+      "per_page" => "1000",
+      "error" => nil,
+      "search" => nil,
+      "sort" => {
+        "by" => nil,
+        "order" => nil
+      },
+      "results" => []
+    }
+  end
+  let(:empty_response_table) do
+    <<eostring
+---|------|-------------|--------------|--------------|-----------
+ID | NAME | DESCRIPTION | ORGANIZATION | REPOSITORIES | SYNC STATE
+---|------|-------------|--------------|--------------|-----------
+eostring
+  end
+
+  it "lists an organization's products" do
+    params = ["--organization-id=#{org_id}"]
+
+    ex = api_expects(:products, :index, "organization product list") do |par|
+      par['organization_id'] == org_id && par['page'] == 1 && par['per_page'] == 1000
+    end
+    ex.returns(empty_response)
+
+    expected_result = success_result(empty_response_table)
+
+    result = run_cmd(@cmd + params)
+    assert_cmd(expected_result, result)
+  end
+
+  it "lists an organization's products filtered by subscription" do
+    params = ["--organization-id=#{org_id}", "--subscription-id=#{subscription_id}"]
+
+    ex = api_expects(:products, :index, "organization product list") do |par|
+      par['organization_id'] == org_id && par['subscription_id'] == subscription_id
+    end
+    ex.returns(empty_response)
```

Comment:
So you've set your stubs to return the `empty_response`, and then you're checking that the response is the `empty_response_table` for each of these tests. Perhaps you should make this assertion in one separate test that tests the output format and remove the assertion from these tests which are supposed to be testing that the command supports the various options (organization_id, sync_plan_id, etc.).